aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -54,11 +54,29 @@ static void prod_kernel_impl(TensorIterator& iter) {
   });
 }
 
+static void and_kernel_impl(TensorIterator& iter) {
+  binary_kernel_reduce_vec(
+    iter,
+    [=](uint8_t a, uint8_t b) -> uint8_t { return a && b; },
+    [=](Vec256<uint8_t> a, Vec256<uint8_t> b) {
+      // adding the implementation here instead of in vec256_base because it is
```

Comment:
nit: I think this comment could go into a little more depth; it's not immediately obvious why the -1/0 distinction matters.

Reply:
Will edit before land. Thanks!